### PR TITLE
[mds-stream] [mds-web-sockets] Refactor NATS to follow the same structure as Kafka, and switch to nats.ts

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

File: packages/mds-web-sockets/launch_server.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
 import { WebSocketServer } from './server'
 
 WebSocketServer()
+  .then(() => console.log(`WebSockerServer running`))
```
Comment (drtyh2o) [inline packages/mds-web-sockets/launch_server.ts:4]: Should rename this file to just `server.ts` and use it as the container-image entry point for webpack.

File: container-images/mds-web-sockets/webpack.config.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
 import webpack from '@mds-core/mds-webpack-config'
 
-export default webpack.Bundle('./index.ts').UsingDefaultConfig()
+export default webpack.Bundle('../../packages/mds-web-sockets/launch_server.ts').UsingDefaultConfig()
```
Comment (drtyh2o) [inline container-images/mds-web-sockets/webpack.config.ts:3]: 1. Rename `launch_server.ts` to just `server.ts`
2. Change DockerFile to `ENTRYPOINT ["/sbin/tini", "node", "--no-deprecation", "server.js"]`

File: packages/mds-web-sockets/wsServer.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
+import logger from '@mds-core/mds-logger'
```
Comment (drtyh2o) [inline packages/mds-web-sockets/wsServer.ts:1]: Only comment is that we typically avoid mixed case file names. I’d go for `ws-server`

File: packages/mds-stream/nats/agency-stream-nats.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+})
+const deviceProducer = NatsStreamProducer<Device>(`${TENANT_ID}.device`)
+const eventProducer = NatsStreamProducer<VehicleEvent>(`${TENANT_ID}.event`)
+const telemetryProducer = NatsStreamProducer<Telemetry>(`${TENANT_ID}.telemetry`)
+
+export const AgencyStreamNats: AgencyStreamInterface = {
+  initialize: async () => {
+    await Promise.all([deviceProducer.initialize(), eventProducer.initialize(), telemetryProducer.initialize()])
+  },
+  writeEvent: eventProducer.write,
```
Comment (macsj200) [inline packages/mds-stream/nats/agency-stream-nats.ts:17]: We can probably use the `...handlers` pattern here, right?

File: packages/mds-stream/nats/agency-stream-nats.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  writeTelemetry: telemetryProducer.write,
+  writeDevice: deviceProducer.write,
+  shutdown: async () => {
+    await Promise.all([deviceProducer.shutdown(), eventProducer.shutdown(), telemetryProducer.shutdown()])
```
Comment (macsj200) [inline packages/mds-stream/nats/agency-stream-nats.ts:21]: This looks pretty clean, I do wonder what happens if one throws an exception before the others resolve though, will the other services still get shut down?

Comment (drtyh2o) [reply packages/mds-stream/nats/agency-stream-nats.ts:21]: `Promise.allSettled` is supported in Node 12 and TS with the ES2020 target.

Comment (avatarneil) [reply packages/mds-stream/nats/agency-stream-nats.ts:21]: Indeed, we probably ought to be using a `Promise.allSettled()`

Comment (avatarneil) [reply packages/mds-stream/nats/agency-stream-nats.ts:21]: No can haz `Promise.allSettled` with Node 12.14.1 :'( https://kangax.github.io/compat-table/es2016plus/

File: packages/mds-stream/nats/agency-stream-nats.ts (hunk 4 above)
Comment (avatarneil) [reply packages/mds-stream/nats/agency-stream-nats.ts:17]: Probably -- I'm going to leave that for a second refactor just so I can get this PR in.

**Pull request merged**